### ISSUE TITLE
[Multimodal] Add support for multiple images per row

### ISSUE THREAD

**Issue by Innixma** [autogluon/autogluon#2784]
Currently, multimodal only supports 1 image per row.

There are valid use-cases where multiple images per row make sense:

1. videos
2. pictures of different views of an object

These multiple images can also be accompanied by text and tabular data.

## Work-around as of v0.6.2

Currently, a user can work-around this limitation by exploding the image column so that 1 row of N images become N rows of 1 image, with all other features + label being duplicated. At test time, explode the image column, then average the predictions across all exploded rows to get the test-time prediction.

This has several draw-backs:

1. Increases training time, even for models that don't use the image column (such as Tabular models)
2. Increases memory usage of Tabular models linearly
3. Runs the risk of overfitting if the user doesn't specify the validation split, since exploded rows could end up split between train and validation.
4. Averaging isn't always the best approach. For example, the positive signal maybe only needs to be found in 1 image for the result to be positive, but averaging a positive signal with many negative signals might return a negative result.

### Initial Solution

In the short-term, an improvement is to move the image column exploding logic to live internally in MultimodalPredictor. This avoids the user having to know how to explode the image column manually, removes the risk of overfitting, and avoids Tabular models being slowed down by the explode logic.

### Stretch Goal 1

A stretch goal will be to implement logic that supports custom post-processing of the exploded prediction probabilities when calculating the final prediction probability of the original row. For example, we could support:

```
['average', 'positive_power_mean', 'negative_power_mean', 'max', 'min', 'majority_voting']
```

### Stretch Goal 2

A natural extension is to automatically identify the appropriate post-processing mechanism based on whichever results in the best validation score. This would be useful as having the user decide which option to select will be complicated, and many will either be unaware of the logic or unsure which to choose.

Automatically selecting should be computationally cheap, as it only requires post-processing the validation prediction probabilities in a for-loop of different methods and scoring each of them, requiring no additional calls to model inference. The chosen post-processing method is simply the one that achieves the best validation score.

### Stretch Goal 3

A third stretch goal is to incorporate this logic to go beyond the multiple image scenario. What we are really supporting is the grouping of rows such that a group of rows' prediction probabilities are post-processed together to get an overall 'group' prediction probability.

A user may want to do this for an arbitrary grouping of rows. They could pass in an extra `group` parameter that indicates which rows belong to each-other and should have the post-processing averaging logic apply.

**Comment by sxjscience** [comment]
Why's it important?

**Comment by billytcl** [comment]
Hi there, just wanted to ask a quick question to clarify. In the Multimodal FAQ (https://auto.gluon.ai/stable/tutorials/multimodal/multimodal-faq.html) it says:

```
How does AutoGluon MultiModal handle multiple images per sample?
We provide two options:

Utilizing multiple image columns in a dataframe.

Concatenating multiple image paths with semicolons and putting them into one dataframe column.

These options can be used individually or in combination. All the image columns are automatically detected. During processing, we pass all images through a single image backbone and average their features and logits to obtain the final representation or prediction. Note that for option 2, the maximum number of images allowed for one dataframe column can be controlled via hyperparameter timm_image.max_img_num_per_col, which has a default value of 2.


```

What is the extent of current support for multiple images?
